src/dymoprint/dymo_print_engines.py

```diff
@@ -119,6 +119,96 @@ def render_barcode(
         )
         return code_bitmap
 
+    def __create_custom_code_bitmap(self, barcode_input_text, bar_code_type):
+        """
+        Renders a barcode image from the input text and barcode type.
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+
+        Returns:
+            Image: A barcode image.
+        """
+        label_height = DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 8
+        if len(barcode_input_text) == 0:
+            return Image.new("1", (1, label_height))
+
+        code = barcode_module.get(
+            bar_code_type, barcode_input_text, writer=BarcodeImageWriter()
+        )
+
+        code_bitmap = code.render(
+            {
+                "font_size": 10,
+                "vertical_margin": 2,
+                "module_height": (DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 6)
+                - 16,
+                "module_width": 1,
+                "background": "black",
+                "foreground": "white",
+            }
+        )
+        return code_bitmap
+
+    def render_barcode_with_text(
+        self,
+        barcode_input_text,
+        bar_code_type,
+        font_file_name: str,
+        frame_width,
+        font_size_ratio=0.9,
+        align="center",
+    ):
+        """
+        Renders a barcode image with the text below it.
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+            font_file_name (str): The name of the font file to be used.
+            frame_width (int): The width of the frame around the text.
+            font_size_ratio (float): The ratio of font size to line height. Default
+                is 1.
+
+        Returns:
+            Image: A barcode with text image.
+        """
+        assert align in ("left", "center", "right")
+        code_bitmap = self.__create_custom_code_bitmap(
+            barcode_input_text, bar_code_type
+        )
+
+        text_bitmap = self.render_text(
+            barcode_input_text, font_file_name, frame_width, font_size_ratio, align
+        )
+        text_width = int(
+            text_bitmap.width // (text_bitmap.height / (code_bitmap.height / 2))
+        )
+        text_bitmap = text_bitmap.resize((text_width, code_bitmap.height // 2))
+
+        # Merge two images into one bitmap
+        code_with_text_bitmap = Image.new(
+            "RGB", (code_bitmap.width, 3 * code_bitmap.height // 2), "black"
```

Comment:
Is there a particular reason you create an `"RGB"` image here instead of a `"1"` (black-and-white image)? Otherwise it's best to stick to `"1"`.